python/lsst/cp/verify/verifyDefects.py

```diff
@@ -117,25 +279,57 @@ def verify(self, exposure, statisticsDict):
         success : `bool`
             A boolean indicating if all tests have passed.
         """
-        ampStats = statisticsDict['AMP']
+        # Amplifier statistics
+        ampStats = statisticsDict["AMP"]
         verifyStats = {}
-        success = True
+        successAmp = True
         for ampName, stats in ampStats.items():
             verify = {}
 
             # These are not defined in DMTN-101 yet.
-            verify['OUTLIERS'] = bool(stats['UNMASKED_OUTLIERS'] >= stats['OUTLIERS'])
-            verify['STDEV'] = bool(stats['UNMASKED_STDEV'] >= stats['STDEV'])
-            verify['MIN'] = bool(stats['UNMASKED_MIN'] <= stats['MIN'])
-            verify['MAX'] = bool(stats['UNMASKED_MAX'] >= stats['MAX'])
+            verify["OUTLIERS"] = bool(stats["UNMASKED_OUTLIERS"] >= stats["OUTLIERS"])
+            verify["STDEV"] = bool(stats["UNMASKED_STDEV"] >= stats["STDEV"])
+            verify["MIN"] = bool(stats["UNMASKED_MIN"] <= stats["MIN"])
+            verify["MAX"] = bool(stats["UNMASKED_MAX"] >= stats["MAX"])
 
             # This test is bad, and should be made not bad.
-            verify['PROB_TEST'] = bool(stats['STAT_OUTLIERS'] == stats['DEFECT_PIXELS'])
+            verify["PROB_TEST"] = bool(stats["STAT_OUTLIERS"] == stats["DEFECT_PIXELS"])
 
-            verify['SUCCESS'] = bool(np.all(list(verify.values())))
-            if verify['SUCCESS'] is False:
-                success = False
+            verify["SUCCESS"] = bool(np.all(list(verify.values())))
+            if verify["SUCCESS"] is False:
+                successAmp = False
 
             verifyStats[ampName] = verify
 
-        return {'AMP': verifyStats}, bool(success)
+        # Detector statistics
+        detStats = statisticsDict["DET"]
+        verifyStatsDet = {}
+        successDet = True
+        # Cosmic rays test from DM-38563, before and after defects.
+        verifyStatsDet["NUMBER_COSMIC_RAYS"] = bool(
+            detStats["NUM_COSMICS_BEFORE"] > detStats["NUM_COSMICS_AFTER"]
+        )
+
+        verifyStatsDet["SUCCESS"] = bool(np.all(list(verifyStatsDet.values())))
+        if verifyStatsDet["SUCCESS"] is False:
+            successDet = False
+
+        # Catalog statistics
+        catStats = statisticsDict["CATALOG"]
+        verifyStatsCat = {}
+        successCat = True
+        # Detection tests from DM-38563, before and after defects.
+        verifyStatsCat["NUMBER_DETECTIONS"] = bool(
+            catStats["NUM_OBJECTS_BEFORE"] > catStats["NUM_OBJECTS_AFTER"]
+        )
+
+        verifyStatsCat["SUCCESS"] = bool(np.all(list(verifyStatsCat.values())))
+        if verifyStatsCat["SUCCESS"] is False:
+            successCat = False
+
+        success = successDet & successAmp & successCat
```

Comment:
I'm not convinced `successAmp` needs to be included, but if it doesn't cause problems, there's no reason to remove it, I think.